File: matura.lua
function sklej(n)
  if n == 1 then return 1 end
  return n - 1 + sklej(math.ceil(n/2)) + sklej(math.floor(n/2))
end

function sklej_gen()
  tab = {1}
  setmetatable(tab, {
    __index = function(table, n)
      table[n] = n - 1 + table[math.ceil(n/2)] + table[math.floor(n/2)]
      return table[n]
    end
  })
  return tab
end

sklejone = sklej_gen()
for i=1,10000 do
  print(sklejone[i])
end

File: matura.js
var sklej = function(n) {
  if (n == 1)
    return 1;
  else
    return n - 1 + sklej(Math.floor(n/2)) + sklej(Math.ceil(n/2));
}

var sklejGen = function(size) {
  var tab = [1,1];

  for (var i = 2; i < size; i++) {
    tab[i] = i - 1 + tab[Math.floor(i/2)] + tab[Math.ceil(i/2)];
  }

  return tab;
}

var generated = sklejGen(10000);

for (var i in generated)
  console.log(i);
